action.yml

```diff
@@ -0,0 +1,213 @@
+name: 'Build and Push Container Images'
+description: 'Builds images on PRs and builds/pushes to registries on default branch'
+
+inputs:
+  # Image configuration
+  image_name:
+    description: 'Base image name (e.g., myorg/myapp)'
+    required: true
+
+  dockerfile_path:
+    description: 'Path to Dockerfile'
+    required: false
+    default: './Dockerfile'
+
+  build_context:
+    description: 'Build context directory'
+    required: false
+    default: '.'
+
+  build_target:
+    description: 'Docker build target stage'
+    required: false
+    default: ''
+
+  build_args:
+    description: 'Docker build arguments (multiline)'
+    required: false
+    default: ''
+
+  # Registry configuration
+  ghcr:
+    description: 'Enable GitHub Container Registry'
+    required: false
+    default: 'true'
+
+  ghcr_image_name:
+    description: 'GHCR image name (defaults to ghcr.io/{owner}/{image_name})'
+    required: false
+    default: ''
+
+  publish_on_pr:
+    description: 'Publish images on pull requests (SHA tags only)'
+    required: false
+    default: 'false'
+
+  google_ar:
+    description: 'Enable Google Artifact Registry'
+    required: false
+    default: 'false'
+
+  google_ar_image_name:
+    description: 'Google Artifact Registry image name'
+    required: false
+    default: ''
+
+  google_ar_tag_prefix:
+    description: 'Google Artifact Registry latest tag'
+    required: false
+    default: ''
+
+  google_workload_identity_provider:
+    description: 'Google Workload Identity Provider'
+    required: false
+    default: ''
+
+  google_service_account:
+    description: 'Google Service Account'
+    required: false
+    default: ''
+
+  # Build configuration
+  platform:
+    description: 'platform to build (e.g., amd64, arm64)'
+    required: false
+    default: 'amd64'
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Setup build metadata
+      id: setup
+      shell: bash
+      run: |
+        # Setup image names
+        GHCR_IMAGE_NAME="${{ inputs.ghcr_image_name }}"
```

Comment:
Should we validate that `inputs.ghcr_image_name` has a format of `ghcr.io/foo/bar`, or simply starts with `ghcr.io/`

Reply:
For most of the cases yes. Reason I put this here is because in sentry, for a long time we were using `ghcr.io/getsentry/sentry-self-hosted` instead of `ghcr.io/getsentry/sentry` but if we're going to standardize makes much more sense to always use `ghcr.io/<owner>/<repo_name>`